### DISCUSSION TITLE
Postgresql access from Nuxt3

### DISCUSSION THREAD

**Discussion by kennerstross** [nuxt/nuxt#17548] [Questions] [answered]
Has anyone successfully connected to postgresql from within Nuxt3? I have not been able to do so myself, and I have not yet found an example of doing so with Nuxt3.

I've tried 2 packages: node-postgres (i.e., pg), but it generates this error:

`node_modules/pg/lib/native/client.js:4:21: error: Could not resolve "pg-native" (mark it as external to exclude it from the bundle, or surround it with try/catch to handle the failure at run-time)`

All of the suggested solutions to this are about overriding some webpack logic, and I have not found any way to apply those suggestions to Nuxt3.

I've also tried a library called ts-postgres, but it depends on process.env with statements such as:

`export const host = process.env.PGHOST || 'localhost';
`

These fail because process.env is undefined. The new approach to handling .env doesn't apply here, as this is a third-party library and I can't get it to access nuxt configuration data.

Has anyone managed to connect to postgresql with Nuxt3? How? With what library? I love Nuxt3 so far, but if I can't use a database, then... well it's a non-starter, obviously. And please don't suggest supabase - this application needs to access a postgres database running on aws (without using an orm, preferably), like thousands of others. Surely this must be possible.

Thanks in advance for your help.

**Comment by kennerstross** [comment, marked as answer]
Ok, I finally got something working with prisma. (Yes, it's an orm, but it looks better than typeorm and it has good raw query support.) Instantiation was tricky; this suggestion seems to work:

import Prisma from '@prisma/client'
import { PrismaClient } from '@prisma/client'

export let prisma: PrismaClient

if (Prisma === undefined) {
  import('@prisma/client').then(({ PrismaClient }) => {
    prisma = new PrismaClient()
  })
} else {
  prisma = new Prisma.PrismaClient()
}

**Reply by cawa-93** [reply]
Could you tell more aboute your case? Why you init prisma like that? Does you have some issues with dev server? I try prisma too, but got problems with HMR (https://github.com/nuxt/framework/issues/4696#issuecomment-1127940949)

**Reply by kennerstross** [reply]
Sorry, cawa, but I can't remember what the problems were (and they might not even problems anymore, since I was using a very early version of nuxt3 at that time). All I know is I stumbled across that solution somewhere on the internet and it worked, so I used it. Have you tried this approach?
